### DISCUSSION TITLE
How to get started developing Xournal++

### DISCUSSION THREAD

**Discussion by UvuvDev** [xournalpp/xournalpp#4581] [General]
Hey, I’m looking to take my C++ further and start contributing to some projects, and I like this one. It seems simple enough for me to start developing without being a 20 year expert in C++. Just wondering if you guys had any advice to get started, if there’s anything I should take into account while developing organizationally (so that way I don’t mess things up for everyone else). Thanks

**Comment by rolandlo** [comment]
Great that you plan to contribute to Xournal++! Yes, it's very possible to contribute without being a 20 year expert in C++.

What you should take into account:
- Before starting to put a lot of work in some feature/fixes make sure there is an issue on the [issue tracker](https://github.com/xournalpp/xournalpp/issues/) suggesting the feature/reporting the bug. You will get some feedback there if the feature is welcome/if the perceived bug is something that needs to be fixed and if someone is already working on it.
- Start with small contributions, as it is easier to review. If you plan on a bigger contribution think about whether it could be split into several smaller pieces (PR's). 
- Read https://github.com/xournalpp/xournalpp/blob/master/CONTRIBUTING.md for more info.

Besides that let me point out that some labels put on the issues (like "good first issue", "priority::high" and "priority::medium") can help find a good task as well. See also https://github.com/xournalpp/xournalpp/issues/2354 for a partial list of tasks.
